rita/src/client.rs

```diff
@@ -262,13 +264,74 @@ fn main() {
             .route("/wifi_settings/ssid", Method::POST, set_wifi_ssid)
             .route("/wifi_settings", Method::GET, get_wifi_config)
             .route("/wipe", Method::POST, wipe)
-    }).workers(1)
-    .bind(format!(
-        "[::0]:{}",
-        SETTING.get_network().rita_dashboard_port
-    )).unwrap()
-    .shutdown_timeout(0)
-    .start();
+    };
+
+    match (
+        SETTING.get_network().key_path.clone(),
+        SETTING.get_network().cert_path.clone(),
+    ) {
+        (Some(key), Some(cert)) => {
+            if path::Path::new(&key).exists() && path::Path::new(&cert).exists() {
+                info!(
+                    "TLS: Starting HTTPS-enabled dashboard (key: {}, cert: {})",
+                    key, cert
+                );
+                let mut builder = SslAcceptor::mozilla_intermediate(SslMethod::tls()).unwrap();
+                builder
+                    .set_private_key_file(
+                        SETTING
+                            .get_network()
+                            .key_path
+                            .clone()
+                            .unwrap_or("/etc/uhttpd.key".to_owned()),
```

Comment:
ok this is not a good way to do defaults. Just crash if the configured path is not good. We should separate bootstrapping from configuration concerns.